### PR TITLE
python3Packages.numpy-groupies: init at 0.11.2

### PR DESCRIPTION
**Pull request by berquist**
## Description of changes

This Python package is a dependency of a dependency for https://github.com/scverse/anndata, a part of https://github.com/NixOS/nixpkgs/issues/313389.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
  - no dependents
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - no binary files
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/numpy-groupies/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    setuptools-scm
+  ];
+
+  dependencies = [ numpy ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    numba
+    pandas
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/numpy-groupies/default.nix:36]:
```suggestion
  ];
  
  pythonImportsCheck = [ "numpy_groupies" ];
```
Could you add `pythonImportsCheck` for a smoke test?

> pythonImportsCheck is set. This is still a good smoke test even if pytestCheckHook is set.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

File: pkgs/development/python-modules/numpy-groupies/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/ml31415/numpy-groupies";
```
Comment (natsukium) [inline pkgs/development/python-modules/numpy-groupies/default.nix:39]:
```suggestion
    homepage = "https://github.com/ml31415/numpy-groupies";
    changelog = "https://github.com/ml31415/numpy-groupies/releases/tag/v${version}";
```
Adding `meta.changelog` is useful for future updates.

File: pkgs/development/python-modules/numpy-groupies/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "numpy-groupies";
+  version = "0.11.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ml31415";
+    repo = pname;
```
Comment (natsukium) [inline pkgs/development/python-modules/numpy-groupies/default.nix:20]:
```suggestion
    repo = "numpy-groupies";
```
This is a misuse of `pname`. `repo` and `pname` are semantically different.

**Comment by natsukium** [comment]
@ofborg build python312Packages.numpy-groupies python311Packages.numpy-groupies

**Review by natsukium — APPROVED** [review]
LGTM, thank you.
Welcome to nixpkgs!🎉

File: pkgs/development/python-modules/numpy-groupies/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    numba
+    pandas
+  ];
+
+  pythonImportsCheck = [ "numpy_groupies" ];
+
+  passthru.updateScript = gitUpdater { tagPrefix = "v"; };
```
Comment (zowoq) [inline pkgs/development/python-modules/numpy-groupies/default.nix:41]: This update script doesn't eval, which broke part of the `nixpkgs-update` bot that tries to run every updateScript.

```
$ nix-shell maintainers/scripts/update.nix --argstr package python3Packages.numpy-groupies
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nixpkgs-update-script'
         whose name attribute is located at /home/zowoq/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:334:7

       … while evaluating attribute 'shellHook' of derivation 'nixpkgs-update-script'

         at /home/zowoq/nixpkgs/maintainers/scripts/update.nix:229:3:

          228|   '';
          229|   shellHook = ''
             |   ^
          230|     unset shellHook # do not contaminate nested shells

       error: function 'anonymous lambda' called with unexpected argument 'tagPrefix'

       at /home/zowoq/nixpkgs/pkgs/common-updater/git-updater.nix:6:1:

            5|
            6| { pname ? null
             | ^
            7| , version ? null

```

Comment (natsukium) [reply pkgs/development/python-modules/numpy-groupies/default.nix:41]: Sorry for the breakage.
I've fixed it by https://github.com/NixOS/nixpkgs/pull/334084, thank you.

**Pull request merged**